agent/consul/leader_connect_test.go

```diff
@@ -177,7 +177,8 @@ func waitForActiveCARoot(t *testing.T, srv *Server, expect *structs.CARoot) {
 func TestLeader_SecondaryCA_IntermediateRenew(t *testing.T) {
 	t.Parallel()
 
-	intermediateCertRenewInterval = time.Millisecond
+	structs.IntermediateCertRenewInterval = time.Millisecond
```

Comment:
Since you are changing globals, you should defer a revert of the value back to what it was before the change. Also you probably should NOT have `t.Parallel()` with this test due to the global manipulation.

Reply:
Good catch! Changed it!